src/main/java/spark/Spark.java

```diff
@@ -36,11 +31,24 @@
  *
  * @author Per Wendel
  */
-public final class Spark extends SparkBase {
+public final class Spark {
+    private static SparkInstance instance;
+
     // Hide constructor
     private Spark() {
     }
 
+    /**
+     * Initializes singleton.
+     */
+    private static class SingletonHolder {
```

Comment:
It's one of many Singleton idioms. I normally use this one, that's the reason.
https://en.wikipedia.org/wiki/Singleton_pattern#Initialization-on-demand_holder_idiom